### PR TITLE
fix #13764 set[T] construction is not typechecked for variables

### LINKED ISSUE #11804: Compile-time illegal conversion from int32 to [0..65535] on set `contains`
Passing an int32 out of int16 range seems to work a runtime but it does not at compiletime.

### Example

```nim
proc fooFunc(c: int): bool =
  result = c in {'a'.ord .. 'z'.ord}

static:
  var c = 194708
  echo fooFunc(c)
```

### Current Output
```
Error: illegal conversion from '194708' to '[0..65535]'
```

### Expected Output

```
false
```

### Additional Information

```
Nim Compiler Version 0.20.99 [Linux: amd64]
Compiled at 2019-07-18
Copyright (c) 2006-2019 by Andreas Rumpf

active boot switches: -d:release
```

### LINKED ISSUE #13764: set[T] construction is not typechecked for variables
This bug caused erroneously written code to compile and execute if the value inside the variables were small enough.

### Example
```nim
#var a: set[int] # Error: set is too large

var b = 1_000_000
let a = {b} # Compiles
```

### Current Output
Compiles, crashes at runtime (expected). If you change `b` into a number smaller than `high(uint8)`, then no error occurs.

### Expected Output
Error about the type of the set being too big? `b` was resolved to be `int` before the set was even made.

### Additional Information
```
$ nim -v
Nim Compiler Version 1.1.1 [Linux: amd64]
Compiled at 2020-02-24
Copyright (c) 2006-2019 by Andreas Rumpf

git hash: 3dad130034e3212159dda5d61988ca2b18959b24
active boot switches: -d:release -d:useLinenoise -d:nativeStackTrace
```

### PR DESCRIPTION
**Pull request by bung87**
fix #13764
fix https://github.com/nim-lang/Nim/issues/11804

refs: https://github.com/nim-lang/Nim/issues/17974

### CODE REVIEW

**Comment by bung87** [comment]
CI fails shows nimquery, balls rely on runtime value. as well as nim std libs ...

refs: https://github.com/GULPF/nimquery/issues/20 done

refs: https://github.com/disruptek/balls/issues/45 done

refs: https://github.com/nim-lang/db_connector/pull/1 done

refs: https://github.com/genotrance/snip/pull/15 done

refs: https://github.com/status-im/nim-stew/pull/163 done

refs: https://github.com/nim-lang/fusion/pull/105 done

refs: https://github.com/status-im/nim-chronos/pull/341 done

refs: https://github.com/Vindaar/ggplotnim/issues/163 done

refs: https://github.com/nitely/nim-unicodedb/pull/25 done

refs: https://github.com/nitely/nim-regex/pull/117 done

refs: https://github.com/nim-lang/bigints/pull/120 done but havn't bump new version

refs: https://github.com/treeform/pixie/pull/513 done

refs: https://github.com/ringabout/httpx/issues/14 done

refs: https://github.com/nitely/nim-unicodedb/issues/23 done

refs: https://github.com/mratsim/Arraymancer/issues/574 done

refs: https://github.com/dom96/httpbeast/issues/81 done but havn't bump new version

File: compiler/semexprs.nim [hunk 1]
```diff
@@ -2521,6 +2521,11 @@ proc semSetConstr(c: PContext, n: PNode, expectedType: PType = nil): PNode =
       typ = makeRangeType(c, 0, MaxSetElements-1, n.info)
     elif lengthOrd(c.config, typ) > MaxSetElements:
       typ = makeRangeType(c, 0, MaxSetElements-1, n.info)
+    if expectedElementType.sym != nil and 
+        sfImportc notin expectedElementType.sym.flags and
+        lengthOrd(c.config, expectedElementType) > MaxSetElements and
+        not (expectedElementType.isIntLit and expectedElementType.n.intVal <= MaxSetElements):
+      localError(c.config, n.info, errSetTooBig)
```
Comment (Araq) [inline compiler/semexprs.nim:2569]: This rule makes no sense. The real check seems to be just `lengthOrd(c.config, expectedElementType) > MaxSetElements`

File: compiler/ccgstmts.nim [hunk 2]
```diff
@@ -654,7 +654,7 @@ proc genParForStmt(p: BProc, t: PNode) =
     #initLoc(forLoopVar.loc, locLocalVar, forLoopVar.typ, onStack)
     #discard mangleName(forLoopVar)
     let call = t[1]
-    assert(call.len in {4, 5})
+    assert(call.len == 4 or call.len == 5)
```
Comment (beef331) [inline compiler/ccgstmts.nim]: `call.len in 4..5`

File: compiler/commands.nim [hunk 3]
```diff
@@ -866,7 +866,7 @@ proc processSwitch*(switch, arg: string, pass: TCmdLinePass, info: TLineInfo;
   of "verbosity":
     expectArg(conf, switch, arg, pass, info)
     let verbosity = parseInt(arg)
-    if verbosity notin {0..3}:
+    if verbosity.uint8 notin {0'u8..3'u8}:
```
Comment (beef331) [inline compiler/commands.nim]: `verbosity notin 0..3`

File: compiler/semtypes.nim [hunk 4]
```diff
@@ -1820,7 +1820,7 @@ proc semTypeNode(c: PContext, n: PNode, prev: PType): PType =
       result = semTypeExpr(c, n, prev)
     else:
       let op = considerQuotedIdent(c, n[0])
-      if op.id in {ord(wAnd), ord(wOr)} or op.s == "|":
+      if op.id == ord(wAnd) or op.id == ord(wOr) or op.s == "|":
```
Comment (beef331) [inline compiler/semtypes.nim]: `op.id in [ord(wAnd), ord(wOr)] or op.s == "|"`

File: compiler/suggest.nim [hunk 5]
```diff
@@ -209,7 +209,7 @@ proc `$`*(suggest: Suggest): string =
     result.add(sep)
     when defined(nimsuggest) and not defined(noDocgen) and not defined(leanCompiler):
       result.add(suggest.doc.escape)
-    if suggest.version in {0, 3}:
+    if suggest.version == 0 or suggest.version == 3:
```
Comment (beef331) [inline compiler/suggest.nim]: `suggest.version in [0, 3]`

**Comment by bung87** [comment]
as of https://github.com/nim-lang/Nim/commit/b0a98cc01e14a33d75866c10c290f63031dc2112 this may not needed.

**Pull request closed without merging**